head.go

```diff
@@ -228,6 +228,50 @@ func (h *headBlock) Busy() bool {
 	return atomic.LoadUint64(&h.activeWriters) > 0
 }
 
+// SeriesLatest is series with its latest value.
+type SeriesLatest struct {
+	Lset      labels.Labels
+	LastValue float64
+}
+
+func (h *headBlock) SelectLatest(cutoff int64, matchers ...labels.Matcher) []SeriesLatest {
```

Comment:
As there can be arrow per my other comment, this also needs to return an errors.
It should probably follow convention of other interfaces and return an iterator rather than a materialized slice.

For example, a federation query probably should not but generally may return all series. With 5+ million series, we don't really want to allocate all that space just to subsequently iterate over it to write out on an HTTP connection.